acceptable/_service.py

```diff
@@ -69,26 +110,37 @@ def bind(self, flask_app):
 class AcceptableAPI:
     """Metadata abount an api endpoint."""
 
-    def __init__(self, service, name, url, options):
-        self.service = service
+    def __init__(self, name, url, introduced_at, options={}):
         self.name = name
         self.url = url
+        self.introduced_at = introduced_at
         self.options = options
-        self.introduced_at = None
         self.view_fn = None
+        self.request_schema = None
+        self.response_schema = None
+
+    @property
+    def methods(self):
+        return tuple(self.options.get('methods', ('GET',)))
 
     def view(self, introduced_at):
         assert self.view_fn is None, 'api already has view registered'
 
-        def wrapper(fn):
-            self.register_view(introduced_at, fn)
+        def decorator(fn):
+            self.register_view(fn, introduced_at)
             return fn
-        return wrapper
+        return decorator
 
-    def register_view(self, introduced_at, view_fn):
+    def register_view(self, view_fn, introduced_at):
```

Comment:
Seems optimistic (isn't this technically public API?), though admittedly I can't find any actual users.

Reply:
It's not a public api, it's only used internally. And introduced_at is going to become optional on the view decorator in future, hence moving it around.

Reply:
I did double check all our projects, and no one is using this api, as is the case for a few others, so I'm planning to switch it out now while we can.